Bots/DotNet/Consumers/CodeFirst/SimpleHostBot-2.1/Dialogs/SetupDialog.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Bot.Builder.Dialogs.Choices;
+using Microsoft.Bot.Builder.Skills;
+using Microsoft.Bot.Schema;
+using Microsoft.BotFrameworkFunctionalTests.SimpleHostBot21.Bots;
+
+namespace Microsoft.BotFrameworkFunctionalTests.SimpleHostBot21.Dialogs
+{
+    /// <summary>
+    /// The setup dialog for this bot.
+    /// </summary>
+    public class SetupDialog : ComponentDialog
+    {
+        private readonly IStatePropertyAccessor<string> _deliveryModeProperty;
+        private readonly IStatePropertyAccessor<BotFrameworkSkill> _activeSkillProperty;
+        private readonly SkillsConfiguration _skillsConfig;
+
+        public SetupDialog(ConversationState conversationState, SkillsConfiguration skillsConfig)
+            : base(nameof(SetupDialog))
+        {
+            _skillsConfig = skillsConfig ?? throw new ArgumentNullException(nameof(skillsConfig));
+
+            _deliveryModeProperty = conversationState.CreateProperty<string>(HostBot.DeliveryModePropertyName);
+            _activeSkillProperty = conversationState.CreateProperty<BotFrameworkSkill>(HostBot.ActiveSkillPropertyName);
+
+            // Define the setup dialog and its related components.
+            // Add ChoicePrompt to render available skills.
+            AddDialog(new ChoicePrompt(nameof(ChoicePrompt)));
+
+            // Add main waterfall dialog for this bot.
+            var waterfallDialog = new WaterfallStep[]
```

Comment:
Consider renaming the variable to waterfallSteps for clarity (it is not a dialog, just an array of steps)